.github/workflows/phpunit.yml

```diff
@@ -0,0 +1,111 @@
+
+name: Run PHPUnit Tests
+
+# Controls when the workflow will run.
+# Run this workflow every time a new commit is pushed to your repository
+on: [push, pull_request]
+
+jobs:
+  # This key is the name of the job.
+  run-tests:
+    # The type of system that the job will run on.
+    runs-on: ubuntu-latest
+    # Matrix Build for this job.
+    strategy:
+      matrix:
+        php-versions: ['7.1', '7.2']
+    # Name the matrix build so we can tell them apart.
+    name: PHP Unit Testing of Analyzed Phenotypes Module (PHP ${{ matrix.php-versions }})
+
+    # Service containers to run with `run-tests`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres
+        env:
+          POSTGRES_USER: tripaladmin
+          POSTGRES_PASSWORD: somesupersecurepassword
+          POSTGRES_DB: testdb
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+
+    steps:
+    # 1) Checkout the repository and setup workspace.
+    - uses: actions/checkout@v2
+
+    # 2) Setup PHP according to the version passed in.
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-versions }}
+        extensions: mbstring, intl, php-pgsql, php-gd, php-xml
+        ini-values: memory_limit=2G
+        coverage: xdebug
+        tools: composer, phpunit
+
+    # 3) Install Drush/Drupal/Tripal
+    - name: Setup Drush, Drupal 7.x, Tripal 3.x
+      id: tripalsetup
+      uses: tripal/setup-tripal-action@7.x-3.x-1.0
+      with:
+        postgres_user: tripaladmin
+        postgres_pass: somesupersecurepassword
+        postgres_db: testdb
+
+    # 4) Install Libraries
+    - name: Install Libraries
+      id: installlibraries
+      env:
+        DRUSH: ${{ steps.tripalsetup.outputs.drush_path }}
+        DRUPAL_ROOT: ${{ steps.tripalsetup.outputs.drupal_root }}
+      run: |
+        cd $DRUPAL_ROOT/sites/all/libraries
+        drush pm-download dragndrop_upload
+        git clone https://github.com/SystemDevil/PHP_XLSXWriter_plus
+        wget https://github.com/d3/d3/releases/download/v3.5.14/d3.zip && unzip d3.zip
+        cd $DRUPAL_ROOT/sites/all/libraries && mkdir d3 && mv d3.js d3/ && mv d3.min.js d3/
+
```

Comment:
I would enable the modules installed in each step before moving on to the next. PS. You're seeing the libraries issue because the library module needs to be installed separately and before the others.